altanalyze3/components/annotation/ensembl_biomart.py

```diff
@@ -0,0 +1,503 @@
+"""
+This is a generalized python module for getting data from Ensemble using Biomart server.
+"""
+
+from __future__ import absolute_import, division, print_function
+import requests
+
+from future.utils import native_str
+from builtins import *
+from xml.etree import ElementTree
+import pandas as pd
+from io import StringIO
+from xml.etree.ElementTree import fromstring as xml_from_string
+
+
+DEFAULT_HOST = 'http://www.biomart.org'
+DEFAULT_PATH = '/biomart/martservice'
+DEFAULT_PORT = 80
+DEFAULT_SCHEMA = 'default'
+
+
+class ServerBase(object):
+    """Base class that handles requests to the biomart server.
+    Attributes:
+        host (str): Host to connect to for the biomart service.
+        path (str): Path to the biomart service on the host.
+        port (str): Port to connect to on the host.
+        url (str): Url used to connect to the biomart service.
+        use_cache (bool): Whether to cache requests to biomart.
+    """
+
+    def __init__(self, host=None, path=None, port=None):
+        """ServerBase constructor.
+        Args:
+            host (str): Url of host to connect to.
+            path (str): Path on the host to access to the biomart service.
+            port (int): Port to use for the connection.
+            use_cache (bool): Whether to cache requests.
+        """
+        # Use defaults if arg is None.
+        host = host or DEFAULT_HOST
```

Comment:
If you use None as the default value, it's better to check against None
`host = DEFAULT_HOST if host is None else host`

Reply:
Yep, but when you call this function without `host` parameter provided the default value will be None.

https://python-patterns.guide/python/sentinel-object/